src/utils/hooks/api/use-get-currencies.tsx

```diff
@@ -0,0 +1,77 @@
+import { CurrencyExt, InterbtcPrimitivesCurrencyId, tokenSymbolToCurrency } from '@interlay/interbtc-api';
+import {
+  InterBtc, // on Polkadot
+  Interlay, // On Polkadot
+  KBtc, // on Kusama
+  Kintsugi, // On Kusama
+  Kusama, // on Kusama
+  Polkadot // on Polkadot
+} from '@interlay/monetary-js';
+import { useCallback } from 'react';
+import { useQuery, UseQueryResult } from 'react-query';
+
+const NATIVE_CURRENCIES: Array<CurrencyExt> = [Polkadot, InterBtc, Interlay, KBtc, Kintsugi, Kusama];
+
+type UseGetCurrenciesResult = UseQueryResult<Array<CurrencyExt>> & {
+  getCurrencyFromTicker: (ticker: string) => CurrencyExt;
+  getForeignCurrencyFromId: (id: number) => CurrencyExt;
+  getCurrencyFromIdPrimitive: (currencyPrimitive: InterbtcPrimitivesCurrencyId) => CurrencyExt;
+};
+
+const getCurrencies = async (): Promise<Array<CurrencyExt>> => {
+  const foreignCurrencies = await window.bridge.assetRegistry.getForeignAssets();
+  return [...NATIVE_CURRENCIES, ...foreignCurrencies];
+};
+
+// Returns all currencies, both native and foreign and helping utils to get CurrencyExt object.
+const useGetCurrencies = (bridgeLoaded: boolean): UseGetCurrenciesResult => {
+  const queryResult = useQuery({ queryKey: 'getCurrencies', queryFn: getCurrencies, enabled: bridgeLoaded });
+
+  const getCurrencyFromTicker = useCallback(
+    (ticker: string): CurrencyExt => {
+      if (queryResult.data === undefined) {
+        throw new Error('useGetCurrencies: Cannot call `getCurrencyFromTicker` before currencies are loaded.');
+      }
+
+      const targetCurrency = queryResult.data.find((currency) => currency.ticker === ticker);
+      if (targetCurrency === undefined) {
+        throw new Error(`useGetCurrencies: getCurrencyFromTicker: Currency with ticker ${ticker} not found.`);
+      }
+
+      return targetCurrency;
+    },
+    [queryResult]
+  );
+
+  const getForeignCurrencyFromId = useCallback(
```

Comment:
Is there a reason for handling errors outside of `useQuery`?

Reply:
Yes, this helper function can throw also when the query is successful. For example: in case that id passed to `getForeignCurrencyFromId ` is not id of any foreign asset known to parachain this function will throw even though the query to get foreign currencies was successful.

Reply:
Thanks :) all good then. Could you add a comment explaining this? It's slightly outside of our usual approach (with good reason!) so would be nice to have it explained in the code.